HikVisionSource/src/test/java/io/vantiq/extsrc/HikVisionSource/TestHikVisionConfig.java

```diff
@@ -0,0 +1,252 @@
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vantiq.extjsdk.ExtensionServiceMessage;
+
+public class TestHikVisionConfig extends TestHikVisionBase {
+
+    HikVisionHandleConfiguration handler;
+    
+    NoSendHikVisionCore nCore;
+    
+    String sourceName;
+    String authToken;
+    String targetVantiqServer;
+    
+    Map<String, String> general;
+    
+    @Before
+    public void setup() {
+        sourceName = "src";
+        authToken = "token";
+        targetVantiqServer = "dev.vantiq.com";
+        nCore = new NoSendHikVisionCore(sourceName, authToken, targetVantiqServer);
+        handler = new HikVisionHandleConfiguration(nCore);
+    }
+    
+    @After
+    public void tearDown() {
+        nCore.stop();
+    }
+    
+    @Test
+    public void testEmptyConfig() {
+        Map conf = new LinkedHashMap<>();
+        Map vantiqConf = new LinkedHashMap<>();
+        ArrayList list = new ArrayList();
+        sendConfig(conf, vantiqConf,list);
+        assertTrue("Should fail on empty configuration", configIsFailed());
+    }
+    
+    
+    @Test
+    public void testMissingGeneral() {
+        Map conf = minimalConfig();
+        Map vantiqConf = createMinimalVantiq();
+        ArrayList<Object> cameras = minimalCameraList();
+
+        conf.remove("general");
+        conf.clear();
+        sendConfig(conf, vantiqConf,cameras);
+        assertTrue("Should not fail when missing 'general' configuration", !configIsFailed());
+    }
+    
+    
+    @Test 
+    public void testMissingOptions() {
+        Map conf = minimalConfig();
+        Map vantiqConf = new LinkedHashMap<>();
+        ArrayList<Object> cameras = minimalCameraList();
+        sendConfig(conf, vantiqConf,cameras);
+        assertTrue("Should fail when missing 'options' configuration", configIsFailed());
+    }
+    
+
+    @Test
+    public void testMinimalConfig() {
+        nCore.start(5); // Need a client to avoid NPEs on sends
+        
+        Map conf = minimalConfig();
+        Map vantiqConf = createMinimalVantiq();
+        ArrayList<Object> cameras = minimalCameraList();
+
+        sendConfig(conf, vantiqConf,cameras);
+        assertFalse("Should not fail with minimal configuration", configIsFailed());
+    }
+    @Test
+    public void testAsynchronousProcessingNoCameras() {
+        nCore.start(5);
+        Map conf = minimalConfig();
+        Map vantiqConf = createMinimalVantiq();
+        ArrayList<Object> cameras = minimalCameraList();
+        cameras.clear();
+
+        sendConfig(conf, vantiqConf,cameras);
+        assertFalse("Should fail when No cameras is set correctly", !configIsFailed());
+    }
+    @Test
```

Comment:
blank line before function.  Etc.  I'm not going to mark them all....

Reply:
i have automatic  an formater for that , everything of that kind is fixen i a single keystroke. 
Done